### DISCUSSION TITLE
PHP Fatal error Font not found

### DISCUSSION THREAD

**Discussion by didj0** [dompdf/dompdf#3120] [Q&A] [answered]
Hi.
With all fonts "DejaVu*" I get this error:
```
PHP Fatal error:  Uncaught FontLib\\Exception\\FontNotFoundException: Font not found in:
in /home/admin/web/XXXXXXX/vendor/phenx/php-font-lib/src/FontLib/Font.php:28
\nStack trace:
\n#0 /home/admin/web/XXXXXXX/vendor/dompdf/dompdf/lib/Cpdf.php(1192): FontLib\\Font::load()
\n#1 /home/admin/web/XXXXXXX/vendor/dompdf/dompdf/lib/Cpdf.php(942): Dompdf\\Cpdf->processFont()
\n#2 /home/admin/web/XXXXXXX/vendor/dompdf/dompdf/lib/Cpdf.php(3206): Dompdf\\Cpdf->o_font()
\n#3 /home/admin/web/XXXXXXX/vendor/dompdf/dompdf/src/Adapter/CPDF.php(932): Dompdf\\Cpdf->output()
\n#4 /home/admin/web/XXXXXXX/vendor/dompdf/dompdf/src/Dompdf.php(890): Dompdf\\Adapter\\CPDF->output()
\n#5 /home/admin/web/XXXXXXX/test/index.php(178): Dompdf\\Dompdf->output()
\n#6 {main}
\n  thrown in /home/admin/web/XXXXXXX/vendor/phenx/php-font-lib/src/FontLib/Font.php on line 28
```

version 2.0.1
Ubuntu 20.04.4 LTS (GNU/Linux 5.4.0-137-generic x86_64)

Fonts are located in the "fonts" folder.
Access to folder "0777"

* There are no such problems on version "0.8.3"

Who knows what the reason is?

**Comment by bsweeney** [comment]
Have you set any options that might impact where Dompdf is looking for the font? Is this an upgrade or a fresh install? If it's an upgrade you can try clearing the generated font data (e.g., delete dompdf_font_family_cache.php). Are you able to see the path that was used for the font?

**Reply by didj0** [reply]
I did not perform any additional actions.
Fresh installation 2 0 1
The file "dompdf_font_family_cache.php" not found. There is a file "installed-fonts.list.json"

**Reply by bsweeney** [reply]
dompdf_font_family_cache.php would only be from a previous installation. installed-fonts.dist.json is the information for the fonts that come with Dompdf.

**Comment by didj0** [comment, marked as answer]
Cpdf.php:1186
```
// Write new font
$tmp_name = @tempnam($this->tmp, "cpdf_subset_");
$font_obj->open($tmp_name, BinaryStream::modeReadWrite);
$font_obj->encode(["OS/2"]);
$font_obj->close();
```
**$tmp_name = @tempnam($this->tmp, "cpdf_subset_");**
return empty.

And then there is a call with an empty parameter
:1192
**$font_obj = Font::load($tmp_name);**

**Reply by bsweeney** [reply]
Seems to indicate that the temp directory is not writable by the process running Dompdf. Try setting a value during instantiation:
```
$dompdf = new Dompdf(["tempDir" => "/path/to/writeable/directory"]);
```

**Reply by didj0** [reply]
Yes, that's exactly what I did.
It works.
